pkg/workflow/asset/data/schemas/v1/Workflow.json

```diff
@@ -0,0 +1,94 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "title": "Workflow",
+  "description": "A project nebula workflow definition",
+  "type": "object",
+  "properties": {
+    "apiVersion": {
+      "type": "string",
+      "description": "Nebula workflow schema version",
+      "enum": [
+        "v1"
+      ]
+    },
+    "description": {
+      "type": "string",
+      "description": "An optional user-provided workflow description"
+    },
+    "parameters": {
+      "type": "object",
+      "patternProperties": {
```

Comment:
You don't need to use `patternProperties` here; `additionalProperties` can take a schema for validating anything that isn't otherwise named. i.e.:

```json
{"type": "object", "additionalProperties": {"$ref": "#/definitions/Parameter"}}
```

Reply:
good tip!